Version 0.3.21 isn't built on M1

```
os: macOS
arch: aarch64
C compiler: gcc 11.3.0_1
Fortran compiler: gfortran 11.3.0_1
OpenBLAS version: 0.3.21
```

make command:
```bash
make CC=gcc FC=gfortran HOSTCC=gcc BINARY=64 F_COMPILER=GFORTRAN FEXTRALIB=-lgfortran USE_OPENMP=0 NO_AVX512=1 DYNAMIC_ARCH=1 NUM_THREADS=64
```

On version 0.3.20, the assembly was successful. When updating the library to 0.3.21, I got the following error:
```
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/iamin.c:37:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/imax.c:38:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/min.c:37:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/iamax_thunderx2t99.c:28:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/max.c:37:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/amin.c:37:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
In file included from ../common.h:133,
                 from ../kernel/arm64/../arm/imin.c:36:
/opt/homebrew/Cellar/gcc/11.3.0_1/lib/gcc/11/gcc/aarch64-apple-darwin21/11/include-fixed/math.h:68:5: error: #error "Unsupported value of __FLT_EVAL_METHOD__."
   68 | #   error "Unsupported value of __FLT_EVAL_METHOD__."
      |     ^~~~~
make[5]: *** [isamin_k_NEOVERSEN2.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[5]: *** [smax_k_NEOVERSEN2.o] Error 1
make[5]: *** [isamax_k_NEOVERSEN2.o] Error 1
make[5]: *** [smin_k_NEOVERSEN2.o] Error 1
make[5]: *** [ismax_k_NEOVERSEN2.o] Error 1
make[5]: *** [samin_k_NEOVERSEN2.o] Error 1
make[5]: *** [ismin_k_NEOVERSEN2.o] Error 1
```

--- a/Makefile.arm64
+++ b/Makefile.arm64
@@ -127,7 +127,7 @@ ifeq (1, $(filter 1,$(GCCMINORVERSIONGTEQ4) $(GCCVERSIONGTEQ10)))
 ifneq ($(OSNAME), Darwin)
 CCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2
 else
-CCOMMON_OPT += -march=armv8.5-a+sve -mtune=neoverse-n2
+CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
 endif
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2

--- a/Makefile.arm64
+++ b/Makefile.arm64
@@ -124,7 +124,11 @@ ifeq ($(CORE), NEOVERSEN2)
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ7) $(ISCLANG)))
 ifeq ($(GCCVERSIONGTEQ9), 1)
 ifeq (1, $(filter 1,$(GCCMINORVERSIONGTEQ4) $(GCCVERSIONGTEQ10)))
+ifneq ($(OSNAME), Darwin)
 CCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2
+else
+CCOMMON_OPT += -march=armv8.5-a+sve -mtune=neoverse-n2
+endif
 ifneq ($(F_COMPILER), NAG)
 FCOMMON_OPT += -march=armv8.5-a+sve+sve2+bf16 -mtune=neoverse-n2
 endif

